test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_svm_mem.cpp

```diff
@@ -0,0 +1,413 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "basic_command_buffer.h"
+#include "svm_command_basic.h"
+#include "procs.h"
+
+
+//--------------------------------------------------------------------------
+namespace {
+
+// CL_INVALID_COMMAND_QUEUE if command_queue is not NULL.
+struct CommandBufferCommandSVMQueueNotNull : public BasicSVMCommandBufferTest
+{
+    using BasicSVMCommandBufferTest::BasicSVMCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandSVMMemcpyKHR(
+            command_buffer, queue, svm_out_mem(), svm_in_mem(), data_size(), 0,
+            nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(
+            error, CL_INVALID_COMMAND_QUEUE,
+            "clCommandSVMMemcpyKHR should return CL_INVALID_COMMAND_QUEUE",
+            TEST_FAIL);
+
+        error = clCommandSVMMemFillKHR(command_buffer, queue, svm_in_mem(),
+                                       &pattern_1, sizeof(cl_char), data_size(),
+                                       0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(
+            error, CL_INVALID_COMMAND_QUEUE,
+            "clCommandSVMMemFillKHR should return CL_INVALID_COMMAND_QUEUE",
+            TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    bool Skip() override
+    {
+        return BasicSVMCommandBufferTest::Skip()
+            || is_extension_available(device,
+                                      "cl_khr_command_buffer_multi_device");
+    }
+
+    const cl_char pattern_1 = 0x14;
+};
+
+// CL_INVALID_CONTEXT if the context associated with command_queue,
+// command_buffer, and kernel are not the same.
+struct CommandBufferCommandSVMDifferentContext
+    : public BasicSVMCommandBufferTest
+{
+    using BasicSVMCommandBufferTest::BasicSVMCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandSVMMemcpyKHR(
+            command_buffer, nullptr, svm_out_mem(), svm_in_mem(), data_size(),
+            0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(
+            error, CL_INVALID_CONTEXT,
+            "clCommandSVMMemcpyKHR should return CL_INVALID_CONTEXT",
+            TEST_FAIL);
+
+        error = clCommandSVMMemFillKHR(command_buffer, nullptr, svm_in_mem(),
+                                       &pattern_1, sizeof(cl_char), data_size(),
+                                       0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(
+            error, CL_INVALID_CONTEXT,
+            "clCommandSVMMemFillKHR should return CL_INVALID_CONTEXT",
+            TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    bool Skip() override
+    {
+        return BasicSVMCommandBufferTest::Skip()
+            || is_extension_available(device,
+                                      "cl_khr_command_buffer_multi_device");
```

Comment:
Even with the multidevice extension I think this would still be an error, so don't need to skip with the presence of this extension.

Reply:
@EwanC I will remove Skip() for this test.